components/dashtable_v2.js

```diff
@@ -76,15 +78,17 @@ export default function GlobalDashboardTable(props) {
       activity: studentActivity,
       progress: percentageCompletion,
       detail: (
-        <a
-          // TODO:
-          href={
-            `/dashboard/v2/details/${props.classroomId}/` + `${studentName}`
-          }
+        <Link
+          href={{
+            pathname: '/dashboard/v2/details',
+            query: {
+              studentName: studentName,
+              classroomName: classroomName
+            } // the data
+          }}
```

Comment:
Context for this comment: [Encoding and decoding URL params ](https://www.geeksforgeeks.org/how-to-encode-and-decode-a-url-in-javascript/)

I think [this documentation](https://nextjs.org/docs/app/api-reference/functions/use-search-params) could possibly be helpful in debugging why the query params are showing as encoded, I think you have to decode the params before using.